services/npm/npm-unpacked-size.service.js

```diff
@@ -0,0 +1,65 @@
+import Joi from 'joi'
+import prettyBytes from 'pretty-bytes'
+import { pathParam } from '../index.js'
+import NpmBase, { packageNameDescription } from './npm-base.js'
+
+export default class NpmUnpackedSize extends NpmBase {
+  static category = 'size'
+
+  static route = {
+    base: 'npm/unpacked-size',
+    pattern: ':packageName/:version*',
+  }
+
+  static openApi = {
+    '/npm/unpacked-size/{packageName}': {
+      get: {
+        summary: 'NPM Unpacked Size',
+        parameters: [
+          pathParam({
+            name: 'packageName',
+            example: 'npm',
+            description: packageNameDescription,
+          }),
+        ],
+      },
+    },
+    '/npm/unpacked-size/{packageName}/{version}': {
+      get: {
+        summary: 'NPM Unpacked Size (with version)',
+        parameters: [
+          pathParam({
+            name: 'packageName',
+            example: 'npm',
+            description: packageNameDescription,
+          }),
+          pathParam({
+            name: 'version',
+            example: '4.18.2',
+          }),
+        ],
+      },
+    },
+  }
+
+  async fetch({ packageName, version }) {
+    return this._requestJson({
+      schema: Joi.any(),
```

Comment:
Instead of using `Joi.any()` here, can you write a schema that ensures the data we need exists in the response and is of the right type. There's some docs on input validation in https://github.com/badges/shields/blob/master/doc/input-validation.md . Not tested, but I guess either

```js
Joi.object({
  dist: Joi.object({
    unpackedSize: nonNegativeInteger.required(),
  }).required()
}).required()
```

or

```js
Joi.object({
  dist: Joi.object({
    unpackedSize: nonNegativeInteger
  }).required()
}).required()
```

will be what we want here, depending on how we want to handle the case where `unpackedSize` does not exist. One option would be to accept a missing value and handle it in the code (like you've done). the other option would be to reject the response at the validation stage. I guess that depends on the answer to https://github.com/badges/shields/pull/9954#discussion_r1485606960

Reply:
Could not complete, refer to the review: https://github.com/badges/shields/pull/9954#discussion_r1486738206